packages/block-editor/src/hooks/border-color.js

```diff
@@ -0,0 +1,197 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { addFilter } from '@wordpress/hooks';
+import { getBlockSupport, hasBlockSupport } from '@wordpress/blocks';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import ColorGradientControl from '../components/colors-gradients/control';
+import {
+	getColorClassName,
+	getColorObjectByColorValue,
+} from '../components/colors';
+import useEditorFeature from '../components/use-editor-feature';
+import { BORDER_SUPPORT_KEY } from './border';
+import { cleanEmptyObject } from './utils';
+
+const BORDER_COLOR_SUPPORT_KEY = 'color';
+const EMPTY_ARRAY = [];
```

Comment:
introducing such a variable is not a good idea. When it will be assigned in multiple places it will always point to the same reference, which could cause strange problems. It is better to create empty arrays inline.

Reply:
It depends. If it is passed later as a prop, then it prevents unnecessary re-renders. I see the same pattern used with different color hooks. Unfortunately it isn't documented.

Reply:
I've updated this to pass the colors and gradients to the `ColorGradientControl` via props. This means the control no longer needs to retrieve the colors a second time. The current approach with `EMPTY_ARRAY` does now also prevent those unnecessary re-renders as mentioned.

Reply:
I see. Normally in such cases, when we want to avoid rerenders we would use useRefs. This will also work as expected, but I think it would be good to add a comment to avoid confusion.

Reply:
Thanks, I've added a comment.